File: TDDJavaProject/app/src/main/java/dev/alexferreira/tddproject/ui/view/MainActivity.java
package dev.alexferreira.tddproject.ui.view;

import android.os.Bundle;
import android.widget.TextView;

import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

import butterknife.BindView;
import dev.alexferreira.tddproject.R;
import dev.alexferreira.tddproject.data.model.Receita;
import dev.alexferreira.tddproject.data.repository.ReceitaRepository;
import dev.alexferreira.tddproject.helper.ViewHelper;
import dev.alexferreira.tddproject.ui.contract.MainContract;
import dev.alexferreira.tddproject.ui.presenter.MainPresenter;

public class MainActivity extends BaseView implements MainContract.View, MainContract.Navigation {

    @BindView(R.id.recyclerView)
    RecyclerView recyclerView;
    @BindView(R.id.tv_empty)
    TextView emptyTV;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        presenter = new MainPresenter(new ReceitaRepository(daoCreator.createReceitaDao()));
    }

    @Override
    public void initReceitaList(List<Receita> receitaList) {

    }

    @Override
    public void showList() {
        ViewHelper.showFirstGoneSecond(recyclerView, emptyTV);
    }

    @Override
    public void showEmptyText() {
        ViewHelper.showFirstGoneSecond(emptyTV, recyclerView);
    }

    @Override
    public void openAddReceitaView() {

    }

    @Override
    public void openReceitaView(Receita receita) {

    }
}

File: TDDKotlinProject/app/src/main/java/dev/alexferreira/tddkotlinproject/ui/view/MainActivity.kt
package dev.alexferreira.tddkotlinproject.ui.view

import android.os.Bundle
import android.view.View
import dev.alexferreira.tddkotlinproject.R
import dev.alexferreira.tddkotlinproject.data.model.Receita
import dev.alexferreira.tddkotlinproject.ui.contract.MainContract
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : BaseView<MainContract.View, MainContract.Presenter>(), MainContract.View {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


        fab_add.setOnClickListener {
            presenter.selectAddReceita()
        }
    }

    override fun onStart() {
        super.onStart()
        presenter.onViewStarted(intent)
    }

    override fun initReceitaList(receitaList: MutableList<Receita>?) {

    }

    override fun showList() {
        recyclerView.visibility = View.VISIBLE
        tv_empty.visibility = View.GONE
    }

    override fun showEmptyText() {
        recyclerView.visibility = View.GONE
        tv_empty.visibility = View.VISIBLE
    }

    override fun openAddReceitaView() {
    }

    override fun openReceitaView(receita: Receita?) {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }
}
